services/autoscaling/src/simcore_service_autoscaling/utils/utils_docker.py

```diff
@@ -414,7 +414,14 @@ async def find_node_with_name(
     list_of_nodes = await docker_client.nodes.list(filters={"name": name})
     if not list_of_nodes:
         return None
-    return parse_obj_as(Node, list_of_nodes[0])
+    # note that there might be several nodes with a common_prefixed name. so now we want exact matching
+    list_of_nodes = parse_obj_as(list[Node], list_of_nodes)
+    for node in list_of_nodes:
```

Comment:
MINOR: since you return the node when the condition is met, i suggest

```python
for docker_node in list_of_nodes:
     node = Node.parse_obj(docker_node)
     if node.Description and node.Descripton.Hostname == name
          return node
```

Reply:
these are very few nodes and pydantic might have optimized the parse_obj_as of a list. but will keep it there in case it becomes a problem